### ISSUE TITLE
Gamemode: Standoff

### ISSUE THREAD

**Issue by Lithrun** [AoTTG-2/AoTTG-2#108]
**Is your feature request related to a problem? Please describe.**
Within the standoff, there is a 1 v 1 situation. Each player is locked within a separate area that has 10 titans. There are no refills available. Killing a titan in your area will spawn another titan in the other player’s area. Thereby it is essential to kill titans as fast as you can before you get overrun. The winner is the first player to take out all of the titans in their area.

@AoTTG-2/mapping-team 
- [ ] Develop a map specifically designated for the Standoff gamemode

@AoTTG-2/development-team 
- [x] Implement the Standoff Gamemode
- [ ] Investigate if Standoff Gamemode can be supported on existing maps
- [ ] Investigate on how Standoff Gamemode can be supported on Custom Maps
- [x] Confirm #36 is done. Add a new titan behavior for this gamemode which forces a titan to always go after the standoff specific target

**Comment by miguelgmarques** [comment]
So, i taught on getting an area where it spawns 10 titans, both players starts with 10 points. The winner is the one that reaches 0. Still an early sketch, but its an idea. Everytime a titan is killed, another one spawns in the opposite team area

**Comment by Lithrun** [comment]
In concept it sounds good, although win condition should also be: if the enemy team dies. Make sure that the titan and point limit are settings that can be changed.

**Comment by MmmCrits** [comment]
The gamemode is more or less functional (at least in singleplayer, I can't test it in multplayer), but it's messily implemented. I had to modify spawn service / spawn menu / several other things to get it to work. 

I didn't realise that there is a separate issue for spawn service implementation (#301), so I'm going to finish 301 first because it is going to make some of the spawn service changes I made for 108 outdated.

I'll stop working on this issue and come back when 301 is done.

In case I don't come back to this, here is what still needs to be done:
- Add player titan support.
- Add localization support.
- Add custom options (once 604 is done).
- Ensure the gamemode works in multiplayer.
- Finish any of the TODO comments left.
- See if existing maps can support the standoff gamemode.
- Clean up all the unused/commented out code and remove any Debug.Log comments.

The latest commit is 7874da1b and is on branch #108-Standoff.
